bot/cogs/manage_classes_cog.py

```diff
@@ -266,6 +271,15 @@ async def create_role(self, ctx, class_repr):
         except:
             role = await ctx.guild.create_role(name=class_repr.role, mentionable=False)
         await self.bot.messenger.publish(Events.on_assignable_role_add, role)
+        return role
+
+    async def sync_perms(self, channel, role, cleanup):
+        log.info(f'Syncing channel and role with cleanup')
+        await channel.set_permissions(role, view_channel=True)
+        await channel.set_permissions(cleanup, view_channel=False)
+        role.position=1
+        cleanup.position=0
```

Comment:
i dont believe setting position here updates it in discord?

Reply:
From what I could tell, though it doesn't update it in the list view, it does indeed give the role permission precedence over cleanup. I could not find a way to update the list view.

Reply:
https://discordpy.readthedocs.io/en/v0.16.12/api.html#discord.Client.move_role

Reply:
Done, move_role() wasn't working but edit() worked